### PR TITLE
Maudemor/micro-java new version.

### PR DESCRIPTION
**Pull request by maudemor**
Changed app to be Quarkus based, and to use ConfigProperty to read the env variables.

### CODE REVIEW

**Comment by rsriniva** [comment]
WARNING: Do not merge apps with same name into main/master because existing OCP 4.5 course will break. I understand main is a separate branch from master, but what if in the future github changes alias master to point to main. Until OCP 4.5 classrooms are removed from our ROL catalog - double check this before ANY merges.  
@richardallred -  WDYT?

File: micro-java/pom.xml
Comment (rsriniva) [inline micro-java/pom.xml:108]: will this really matter? you are. already including the quarkus plugin which may take care of build and deployment?

File: micro-java/src/main/docker/Dockerfile.fast-jar
Comment (rsriniva) [inline micro-java/src/main/docker/Dockerfile.fast-jar:24]: excellent!

File: micro-java/pom.xml
Comment (rsriniva) [reply micro-java/pom.xml:108]: it's not clear what the quarkus plugin will handle and what jkube handles. make sure you tell the difference

File: micro-java/src/main/jkube/deployment.yml
Comment (rsriniva) [inline micro-java/src/main/jkube/deployment.yml:29]: prefer specific version rather than use latest tag

**Review by rsriniva — CHANGES_REQUESTED** [review]

**Comment by richardallred** [comment]
> WARNING: Do not merge apps with same name into main/master because existing OCP 4.5 course will break. I understand main is a separate branch from master, but what if in the future github changes alias master to point to main. Until OCP 4.5 classrooms are removed from our ROL catalog - double check this before ANY merges.  
> 
> @richardallred -  WDYT?

AFAIK we should be okay to merge stuff into main since the existing 4.5 material doesn't really reference it at all. I think we can maintain master and main separately until we fully decommission 4.5.

**Comment by maudemor** [comment]
> > WARNING: Do not merge apps with same name into main/master because existing OCP 4.5 course will break. I understand main is a separate branch from master, but what if in the future github changes alias master to point to main. Until OCP 4.5 classrooms are removed from our ROL catalog - double check this before ANY merges.
> > @richardallred -  WDYT?
> 
> AFAIK we should be okay to merge stuff into main since the existing 4.5 material doesn't really reference it at all. I think we can maintain master and main separately until we fully decommission 4.5.

This is my understanding as well.

File: micro-java/src/main/jkube/deployment.yml
Comment (maudemor) [reply micro-java/src/main/jkube/deployment.yml:29]: Good catch. Fixed locally, will push later.

File: micro-java/pom.xml
Comment (maudemor) [reply micro-java/pom.xml:108]: They do handle different things. Would it be ok to add some comments on the `pom.xml`, or should the responsibilities of each plugin be explained in the coursebook?

Comment (rsriniva) [reply micro-java/pom.xml:108]: please add little content, one or two sentences in lecture (not in GE).

Comment (maudemor) [reply micro-java/pom.xml:108]: Like it. Will do so!

**Review by rsriniva — APPROVED** [review]
approved

**Pull request merged**
